### PR TITLE
#120 replace admin PART 1

### PR DESCRIPTION
**Pull request by krios82**
First part of replacing the admin section.
* CRUD for categories and concepts
* R for data elements and admin users

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -28,6 +28,8 @@ gem 'ancestry', '~> 3.0'
 gem 'rails_admin', '~> 1.4'
 gem 'rails_admin_nestable', '~> 0.3.2'
```
Comment (spikeheap) [inline Gemfile:29]: Should this PR remove `rails_admin` as well?

Comment (krios82) [reply Gemfile:29]: I'm keeping that for reference (at the small price of mounting the engine in the routes, I can still look at the existing one for reference).
I was thinking to remove this and all the related files in a separate PR once we've replicated everything it's in now, so:
* PART 3 will be replicating the file upload as it is now, and 
* PART 4 removing this and everything else related to RailsAdmin

Would that be acceptable for you?

**Comment by spikeheap** [comment]
Hi @cacquani, this looks promising. I've tested it locally and have a couple of questions:

1. The admin users list shows a "versions" count, but this doesn't drill down to a history, and increments each time the user logs in. Should we remove that attribute from this first piece of work, and integrate papertrail once this is merged? If not (which is fine, too), could we make this version count relate to material changes to the model, not including log-in counts?

Screenshot 2019-05-01 09 55 49


2. Related to (1): if we're keeping papertrail in this PR, could the versions be listed/linked, or some kind of change history be exposed?

3. "Parent" in the category and concept needs to show the breadcrumb. I'm not familiar with administrate – is this possible?

Screenshot 2019-05-01 10 05 16

4. Within a category or concept, the list of possible children is huge, so the multi-select UI component doesn't work very well. Are there any examples of the search/full list and separate list of selected items we used to have in Rails Admin, or another approach which makes it easier to see the current selection before updating?

Screenshot 2019-05-01 10 06 06

5. Category and concept searches (at the top of each list, e.g. http://localhost:3000/admin/concepts) yields a PostgreSQL error. Can you reproduce this?

Screenshot 2019-05-01 10 06 35

6. Editing your account (clicking your email in the nav bar) isn't styled consistently for me, do you see the same thing?

Screenshot 2019-05-01 10 09 50

7. Is there a plan to add admin users through the interface? It looks like editing/adding is disabled on `UserAdmin` models, but we'll need admins to be able to add additional admins further down the line.

8. Clicking a row in the category/concept/data item list doesn't take you to the page. This requires you to click on the text within the box, which is a bit counter-intuitive given the highlighting. Could we update it to accept a click anywhere within the area?

ezgif com-video-to-gif

**Comment by spikeheap** [comment]
Merging, @cacquani to add the items above as backlog issues

**Pull request merged**